### PR TITLE
Use mkdir -p when creating directories

### PR DESCRIPTION
**Pull request by maage**
#### Description:

This ensure mkdir does not fail if directory already exists.

Also avoid unnecessary tests.

Added simple style changes.

#### Rationale:

Simplify code.

#### Review Hints:

Changes are all over unfortunately.

### CODE REVIEW

**Comment by maage** [comment]
Failures here is good example why something like my #10387 is needed.

Maybe there should be some different test phase / code phrase than "ERROR" to catch these. I think this just means we try to test some rule but it is totally not configured for product. Tests did not fail.
```
2023-05-13T20:40:06.6231004Z ERROR - Rule 'xccdf_org.ssgproject.content_rule_dir_system_commands_root_owned' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-qa6ravy7'
2023-05-13T20:40:06.6287962Z ERROR - Rule 'xccdf_org.ssgproject.content_rule_dir_system_commands_group_root_owned' isn't present in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-8' in '/tmp/ssgts-ds-qa6ravy7'
```

I'll add fix for `mount_option_nodev_nonroot_local_partitions` shortly. But it seems to me that it has never worked in containerized env. My test container has `/dev` with size 64MB and test tries to create 200MB test file to create partition and it fails.

I'll add fix for `dir_perms_world_writable_root_owned` as it did not specially handle `/proc`. And I think it should exclude it.

`dir_perms_world_writable_root_owned` test `world_writable_dir_on_nonlocal_fs.fail.sh` tries to setup NFS server (and `mount_option_nodev_nonroot_local_partitions` `remote_without_nodev.pass.sh`) and it fails in containers because there is no nfsd kernel module and container does not have and most probably never will acquire permissions to load kernel modules. This type of tests should be skipped in containerized envs and tested in virtual environments. And configuration should be per test case as other tests cases might work.

**Comment by maage** [comment]
Well, `mount_option_nodev_nonroot_local_partitions` does not work in containerized env as it uses `mount` and that requires `CAP_SYS_ADMIN` and it is missing and most probably not going to be there anyways.

File: linux_os/guide/system/permissions/files/dir_perms_world_writable_root_owned/bash/shared.sh [hunk 1]
```diff
@@ -1,3 +1,5 @@
 # platform = multi_platform_all
 
-find / -not -fstype afs -not -fstype ceph -not -fstype cifs -not -fstype smb3 -not -fstype smbfs -not -fstype sshfs -not -fstype ncpfs -not -fstype ncp -not -fstype nfs -not -fstype nfs4 -not -fstype gfs -not -fstype gfs2 -not -fstype glusterfs -not -fstype gpfs -not -fstype pvfs2 -not -fstype ocfs2 -not -fstype lustre -not -fstype davfs -not -fstype fuse.sshfs -type d -perm -0002 -uid +0 -exec chown root {} \;
+# At least under containerized env /proc can have files w/o possilibity to
```
Comment (marcusburghardt) [inline linux_os/guide/system/permissions/files/dir_perms_world_writable_root_owned/bash/shared.sh:3]: The content of this commit is also part of https://github.com/ComplianceAsCode/content/pull/10563. Please, remove it here.

File: linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/tests/correct.pass.sh [hunk 2]
```diff
@@ -14,10 +14,10 @@ for partition in ${partitions[@]}; do
     mount -o remount "$partition"
 done
 
-PARTITION="/dev/new_partition1"; create_partition
+PARTITION="/new_partition1"; create_partition
```
Comment (marcusburghardt) [inline linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/tests/correct.pass.sh:20]: I suggest to open another PR for changes in this commit and keep this PR only for changes related to `mkdir -p`.

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
Thanks for this PR @maage . I see the last commits in this PR are not related to `mkdir -p` changes. Please, move these last commits to another PR and keep this only about `mkdir` changes.

File: linux_os/guide/system/permissions/files/dir_perms_world_writable_root_owned/bash/shared.sh (hunk 1 above)
Comment (maage) [reply linux_os/guide/system/permissions/files/dir_perms_world_writable_root_owned/bash/shared.sh]: I removed extra commits

**Review by marcusburghardt — APPROVED** [review]
Thanks for these improvements and also thanks for the great analysis in https://github.com/ComplianceAsCode/content/pull/10556#issuecomment-1546946657.

I confirm the errors reported in the containers tests (automatus) can be waived in this PR.

**Pull request merged**